src/legacy-libraries-migration/data/apiHooks.ts

```diff
@@ -0,0 +1,32 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+
+import * as api from './api';
+
+export const legacyMigrationQueryKeys = {
+  all: ['contentLibrary'],
+  /**
+   * Base key for data specific to a migration task
+   */
+  migrationTask: (migrationId?: string | null) => [...legacyMigrationQueryKeys.all, migrationId],
+};
+
+/**
+ * Use this mutation to update container collections
+ */
+export const useUpdateContainerCollections = () => (
+  useMutation({
+    mutationFn: async (requestData: api.BulkMigrateRequestData) => api.bulkMigrateLegacyLibraries(requestData),
+  })
+);
+
+/**
+ * Get the migration status
+ */
+export const useMigrationStatus = (migrationId: string | null) => (
+  useQuery({
+    queryKey: legacyMigrationQueryKeys.migrationTask(migrationId),
+    queryFn: () => api.getMigrationStatus(migrationId!),
+    enabled: migrationId !== null,
```

Comment:
Since we updated tanstack query to v5, we can use the [skipToken](https://tanstack.com/query/v5/docs/framework/react/guides/disabling-queries#typesafe-disabling-of-queries-using-skiptoken)

```suggestion
    queryFn: migrationId ? () => api.getMigrationStatus(migrationId) : skipToken,
```